src/classes/dataSource.h

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2023 Team Dissolve and contributors
+
+#pragma once
+
+#include "base/serialiser.h"
+#include "io/fileAndFormat.h"
+#include "io/import/data1D.h"
+#include "io/import/data2D.h"
+#include "io/import/data3D.h"
+#include "items/list.h"
+#include "keywords/dataSourceBase.h"
+#include "math/data1D.h"
+#include "math/data2D.h"
+#include "math/data3D.h"
+#include "math/sampledData1D.h"
+#include <variant>
+
+class DataSource
+{
+    public:
+    using DataType = std::variant<Data1D, SampledData1D, Data2D, Data3D>;
+    using Format = std::variant<Data1DImportFileFormat, Data2DImportFileFormat, Data3DImportFileFormat>;
+
+    public:
+    DataSource() = default;
+    ~DataSource() = default;
+
+    public:
+    enum DataSourceType
+    {
+        Internal,
+        External
+    };
+
+    public:
+    // Return enum options for DataSourceType
+    static EnumOptions<DataSourceType> dataSourceTypes();
+    // Return data source type enum
+    DataSourceType dataSourceType() const;
+
+    /*
+     * Data
+     */
+
+    private:
+    // Name of data (tag or filename)
+    std::string dataName_;
+    // Type of data being stored
+    DataSourceType dataSourceType_;
+    // String to hold internal data tag (if internal)
+    std::string internalDataSource_;
+    // Variant to hold file and format of external data (if external)
+    Format externalDataSource_;
+    // Variant to hold data
+    DataType data_;
+
+    public:
+    // Return data name
+    std::string_view dataName() const;
+    // Return if data exists and has been initialised
+    bool dataExists() const;
+    // Return internal data source
+    std::optional<std::string> internalDataSource() const;
+    // Return external data source
+    Format &externalDataSource();
+    // Function to source data (only required for internal data sources)
+    bool sourceData(GenericList &processingModuleData);
+    // Function to add internal data
```

Comment:
Don't really need to describe these functions as functions, so...
```suggestion
    // Source specified data (only required for internal data sources)
    bool sourceData(GenericList &processingModuleData);
    // Set internal data source
```